src/main/java/com/epam/aidial/core/controller/ResourceController.java

```diff
@@ -158,10 +167,23 @@ private Future<?> putResource(ResourceDescription descriptor) {
                     switch (resourceType) {
```

Comment:
Extract a method for validation and transformation.

Reply:
fixed